lib/runtime-c-api/src/lib.rs

```diff
@@ -96,6 +96,7 @@ pub mod global;
 pub mod import;
 pub mod instance;
 pub mod memory;
+pub mod metering;
```

Comment:
You can add the attribute `#[cfg(feature = "metering")]` here, and remove all the same attributes in `metering.rs`. I believe it will simplify the code :-).

Reply:
The problem is that it would expose a different C ABI then, and the go wrapper would fail to compile. Since go doesn't support conditional compilation feature tags as rust does, I think it would break the go bindings to have different ABIs (okay, there is some stuff for architecture, but nothing like rust's feature switches).

I can change that out if this is not a problem, but wanted the least breaking change possible (especially since the abi is committed to the go-ext-wasmer repo)

Reply:
Golang build tags can be used to select the proper C ABI object to link against.